### PR TITLE
Refactor table metadata persist and load logic.

### LINKED ISSUE #20477: Consider refactoring table metadata loading and persistence logic
Hi community,

I think the table metadata persistence and load has some problem. `ShardingSphere` is a distributed database, so we cannot go to the curd governance center based on the local table loading. So, we are plan to split three scenes.

First scenes : After the local and governance center are compared, the persistent table is sent to the governance center.

```
1. REFRESH TABLE METADATA : Refresh the table information of the logic database, ContextManager.reloadDatabase(final String databaseName)

2. REFRESH TABLE METADATA : Refresh the table information of the schema, ContextManager.reloadSchema(final String databaseName, final String schemaName, final String dataSourceName)

3. REFRESH TABLE METADATA : Refresh the table information of the table, ContextManager.reloadTable(final String databaseName, final String schemaName, final String dataSourceName, final String tableName) and ContextManager.reloadTable(final String databaseName, final String schemaName, final String tableName)

```

Second scenes: Local direct persistence to the governance center.

```
1. Governance center has no tables

2. Add resource by DistSQL : ContextManager.updateResources(final String databaseName, final Map<String, DataSourceProperties> toBeUpdatedDataSourcePropsMap)
```

Third scenes: Load table metadata from governance center.

```
1. Governance center has tables

2. Alter resource by DistSQL : ContextManager.updateResources(final String databaseName, final Map<String, DataSourceProperties> toBeUpdatedDataSourcePropsMap)

3. Alter rule configuration by DistSQL : ContextManager.alterRuleConfiguration(final String databaseName, final Collection<RuleConfiguration> ruleConfigs)

4. Event monitoring Add or modify data source : ContextManager.alterDataSourceConfiguration(final String databaseName, final Map<String, DataSourceProperties> dataSourcePropsMap)

5. Toggle new version of metadata : ContextManager.alterDataSourceAndRuleConfiguration(final String databaseName,
                                                                 final Map<String, DataSourceProperties> dataSourcePropsMap, final Collection<RuleConfiguration> ruleConfigs) 
```

### PR DESCRIPTION
**Pull request by zjcnb**
Fixes #20477

### CODE REVIEW

File: shardingsphere-mode/shardingsphere-mode-core/src/main/java/org/apache/shardingsphere/mode/metadata/MetaDataContextsFactory.java [hunk 1]
```diff
@@ -79,4 +83,14 @@ private static DatabaseConfiguration createEffectiveDatabaseConfiguration(final
         Collection<RuleConfiguration> databaseRuleConfigs = persistService.getDatabaseRulePersistService().load(databaseName);
         return new DataSourceProvidedDatabaseConfiguration(effectiveDataSources, databaseRuleConfigs);
     }
+    
+    private static Map<String, ShardingSphereDatabase> reloadDatabases(final Map<String, ShardingSphereDatabase> databases, final MetaDataPersistService persistService) {
+        Map<String, ShardingSphereDatabase> result = new ConcurrentHashMap<>(databases.size(), 1);
```
Comment (taojintianxia) [inline shardingsphere-mode/shardingsphere-mode-core/src/main/java/org/apache/shardingsphere/mode/metadata/MetaDataContextsFactory.java:88]: `Collections.singletonMap()` may make this more elegant ？

Comment (zjcnb) [reply shardingsphere-mode/shardingsphere-mode-core/src/main/java/org/apache/shardingsphere/mode/metadata/MetaDataContextsFactory.java:88]: > `Collections.singletonMap()` may make this more elegant ？

I think you are right when i know the key and value explicitly. But no define value here.

File: shardingsphere-mode/shardingsphere-mode-core/src/main/java/org/apache/shardingsphere/mode/metadata/persist/service/DatabaseMetaDataPersistService.java [hunk 2]
```diff
@@ -159,12 +168,27 @@ public Optional<ShardingSphereSchema> load(final String databaseName, final Stri
...
+    /**
+     * Load schemas.
+     *
+     * @param databaseName database name to be loaded
+     * @return Loaded schemas
+     */
+    public Map<String, ShardingSphereSchema> load(final String databaseName) {
+        Collection<String> schemas = repository.getChildrenKeys(DatabaseMetaDataNode.getMetaDataSchemasPath(databaseName));
+        Map<String, ShardingSphereSchema> result = new ConcurrentHashMap<>(schemas.size(), 1);
+        schemas.forEach(each -> result.put(each.toLowerCase(), load(databaseName, each).orElse(new ShardingSphereSchema())));
```
Comment (TeslaCN) [inline shardingsphere-mode/shardingsphere-mode-core/src/main/java/org/apache/shardingsphere/mode/metadata/persist/service/DatabaseMetaDataPersistService.java:188]: Consider using `orElseGet` here.

Comment (zjcnb) [reply shardingsphere-mode/shardingsphere-mode-core/src/main/java/org/apache/shardingsphere/mode/metadata/persist/service/DatabaseMetaDataPersistService.java]: Ok, done.

**Review by TeslaCN — APPROVED** [review]

**Pull request merged**